backend/models/notification.js

```diff
@@ -156,6 +163,64 @@ module.exports = {
 			.then(c => c.deleteMany({}));
 	},
 
+	createUserNotification: async function (users, teamSpace, modelId, issueId) {
+		return Promise.all(users.map(u => {
+			return this.upsertIssueClosedNotification(u, teamSpace, modelId, issueId)
+				.then((n) => {
+					return ({ username: u, notification: n });
+				});
+		}));
+	},
+
+	removeUserNotification: async function (users, teamSpace, modelId, issueId, issueType) {
+		return Promise.all(
+			users.map(u => {
+				return this.removeIssueNotification(u, teamSpace, modelId, utils.objectIdToString(issueId), issueType)
+					.then((n) => {
+						return Object.assign({ username: u}, n);
+					});
+			})).then(notifications => notifications.reduce((a, c) => !c.notification ? a : a.concat(c), []));
+
+	},
+
+	upsertIssueClosedNotifications: async function (teamSpace, modelId, issue) {
+		const rolesKey = "assigned_roles";
+		const owner = issue.owner;
+		const assignedRoles = new Set();
+
+		const comments = issue.comments;
+
+		for (const item in comments) {
+			const actionProperty = comments[item].action;
+
+			// If no additional roles have been assigned ,
+			// make sure we add the current assigned role.
+			if (actionProperty && actionProperty.property !== rolesKey) {
+				assignedRoles.add(issue.assigned_roles);
+			}
+			// Check for additional roles that have been assigned
+			// using the issue comments.
+
+			// TODO: Add check for .property, should fix travis test fail
+			if (actionProperty && actionProperty.property === rolesKey) {
+				assignedRoles.add(actionProperty.to);
+				assignedRoles.add(actionProperty.from);
```

Comment:
Assuming you always add current assigned roles to the set, you should only ever need to add from here.